src/mame/drivers/iphone2g.cpp

```diff
@@ -206,6 +225,102 @@ iphone2g_timer_device::iphone2g_timer_device(const machine_config &mconfig, cons
+	iphone2g_timers_device(const machine_config &mconfig, const char* tag, device_t *owner, uint32_t clock = 0);
+
+	auto out_irq_cb() { return m_out_irq_func.bind(); }
+
+	void map(address_map &map);
+
+protected:
+	iphone2g_timers_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+
+	// device-level overrides
+	virtual void device_resolve_objects() override;
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual void device_add_mconfig(machine_config &config) override;
+
+	virtual space_config_vector memory_space_config() const override;
+
+private:
+	address_space_config m_mmio_config;
+
+	devcb_write_line m_out_irq_func;
+	required_device_array<iphone2g_timer_device, 7> m_timers;
+
+	u64 ticks;
+};
+
+DECLARE_DEVICE_TYPE(IPHONE2G_TIMERS, iphone2g_timers_device)
+
+void iphone2g_timers_device::map(address_map &map)
+{
+	map(0x00000, 0x00017).m(m_timers[0], FUNC(iphone2g_timer_device::map));
+	map(0x00020, 0x00037).m(m_timers[1], FUNC(iphone2g_timer_device::map));
+	map(0x00040, 0x00057).m(m_timers[2], FUNC(iphone2g_timer_device::map));
+	map(0x00060, 0x00077).m(m_timers[3], FUNC(iphone2g_timer_device::map));
+	map(0x000a0, 0x000b7).m(m_timers[4], FUNC(iphone2g_timer_device::map));
+	map(0x000c0, 0x000d7).m(m_timers[5], FUNC(iphone2g_timer_device::map));
+	map(0x000e0, 0x000f7).m(m_timers[6], FUNC(iphone2g_timer_device::map));
+	map(0x000f8, 0x000fb).lrw32(NAME([this](offs_t offset){ return 0xffffffff; }), NAME([this](offs_t offset, u32 data){ m_out_irq_func(0); }));
+}
+
+device_memory_interface::space_config_vector iphone2g_timers_device::memory_space_config() const
+{
+	return space_config_vector{
+		std::make_pair(0, &m_mmio_config)
+	};
+}
+
+void iphone2g_timers_device::device_resolve_objects()
+{
+	// resolve callbacks
+	m_out_irq_func.resolve_safe();
+}
+
+void iphone2g_timers_device::device_start()
+{
+}
+
+void iphone2g_timers_device::device_reset()
+{
+}
+
+void iphone2g_timers_device::device_add_mconfig(machine_config &config)
+{
+	INPUT_MERGER_ANY_HIGH(config, "timerirq").output_handler().set([this](int state){ if(state) m_out_irq_func(1); });
+	for(int i = 0; i < 6; i++)
+	{
+		IPHONE2G_TIMER(config, m_timers[i], 0);
+	}
+	m_timers[0]->out_irq_cb().set("timerirq", FUNC(input_merger_any_high_device::in_w<0>));
+	m_timers[1]->out_irq_cb().set("timerirq", FUNC(input_merger_any_high_device::in_w<1>));
+	m_timers[2]->out_irq_cb().set("timerirq", FUNC(input_merger_any_high_device::in_w<2>));
+	m_timers[3]->out_irq_cb().set("timerirq", FUNC(input_merger_any_high_device::in_w<3>));
+	m_timers[4]->out_irq_cb().set("timerirq", FUNC(input_merger_any_high_device::in_w<4>));
+	m_timers[5]->out_irq_cb().set("timerirq", FUNC(input_merger_any_high_device::in_w<5>));
+	m_timers[6]->out_irq_cb().set("timerirq", FUNC(input_merger_any_high_device::in_w<6>));
+}
+
+DEFINE_DEVICE_TYPE(IPHONE2G_TIMERS, iphone2g_timers_device, "iphone2g_timers", "iPhone 2G timers")
```

Comment:
It smells more like a _Timer Controller Unit_ to me, assuming it doesn't have an official name. TIMERS is a terrible choice and again prone to replace issues.